src/types/impl-trait.md

```diff
@@ -53,6 +65,16 @@ This includes generic arguments for the return type or any const generics.
 >
 > Therefore, changing the function signature from either one to the other can constitute a breaking change for the callers of a function.
 
+## Abstract types
+
+> Note: these are also called "existential types" or "opaque types".
+
+Abstract types are backed by a "hidden type", but only expose certain traits of that hidden type.
+Every abstract type has exactly one hidden type, and the hidden type is inaccessible (it could even be a private
+type or an unnameable type like a closure). It is required that every abstract type gets a hidden type
+assigned/constrained/bound within its "defining scope" (more to that later). It may get constrained multiple times,
```

Comment:
I think it would be good to link instead of vaguely referring to some other text.

```suggestion
assigned/constrained/bound within its [*defining scope*](#defining-scope). It may get constrained multiple times,
```